src/components/permissionable/permissionable.cairo

```diff
@@ -1,3 +1,112 @@
-// permissionable component
+// *************************************************************************
+//                              PERMISSIONABLE COMPONENT
+// *************************************************************************
+#[starknet::component]
+pub mod PermissionableComponent {
+    // *************************************************************************
+    //                              IMPORTS
+    // *************************************************************************
+    use starknet::storage::{Map, StorageMapReadAccess, StorageMapWriteAccess};
+    use starknet::{ContractAddress, get_caller_address, get_block_timestamp};
+    use token_bound_accounts::components::account::account::AccountComponent;
+    use token_bound_accounts::interfaces::IAccount::{IAccount, IAccountDispatcherTrait};
+    use token_bound_accounts::components::account::account::AccountComponent::InternalImpl;
+    use token_bound_accounts::interfaces::IPermissionable::{
+        IPermissionable, IPermissionableDispatcher, IPermissionableDispatcherTrait
+    };
 
+    // *************************************************************************
+    //                              STORAGE
+    // *************************************************************************
+    #[storage]
+    pub struct Storage {
+        permissions: Map<
+            (ContractAddress, ContractAddress), bool
+        > // <<owner, permissioned_address>, bool>
+    }
 
+    // *************************************************************************
+    //                              EVENTS
+    // *************************************************************************
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    pub enum Event {
+        PermissionUpdated: PermissionUpdated
+    }
+
+    // @notice emitted when permissions are updated for an account
+    // @param owner tokenbound account owner
+    // @param permissioned_address address to be given/revoked permission
+    // @param has_permission returns true if user has permission else false
+    #[derive(Drop, starknet::Event)]
+    pub struct PermissionUpdated {
+        #[key]
+        pub owner: ContractAddress,
+        pub permissioned_address: ContractAddress,
+        pub has_permission: bool,
+    }
+
+    // *************************************************************************
+    //                              ERRORS
+    // *************************************************************************
+    pub mod Errors {
+        pub const UNAUTHORIZED: felt252 = 'Account: unauthorized';
```

Comment:
remove `UNAUTHORIZED`, `NOT_OWNER`, `NOT_PERMITTED` errors, as they are not used anywhere in this component